### ISSUE TITLE
[bug] Retain setting in MQTT does not work

### ISSUE THREAD

**Issue by jasonsinclair5** [OpenZWave/Zwave2Mqtt#515]
**Version**

Build/Run method
- [x] Docker
- [ ] PKG
- [ ] Manually built (git clone - npm install - npm run build ) 

Zwave2Mqtt version: 3.2.1
Openzwave Version: 1.6.1123

**Describe the bug**
Retain and retained discovery does not seem to hold. Upgraded from 3.1.0, now HA is showing devices on when they're off, and using MQTT explorer, no topics have the "Retained" marking. Checked other devices (Tasmota, Dafang) that are using retained, and they are all working. I tried to disable the gateway, turn off retain and retain discovery, re-enable, delete topics in MQTT manually, and rediscover. The topics show up again in MQTT, but still aren't retained.

**Expected behavior**
Retain setting to work as it did in previous versions.

**Additional context**
None.

**Comment by robertsLando** [comment]
@jasonsinclair5 I cannot reproduce your problem. @chrisns did you have problems with retain?

**Comment by jasonsinclair5** [comment]
> @jasonsinclair5 I cannot reproduce your problem. @chrisns did you have problems with retain?

After reviewing this issue further, I don’t think it’s the retain setting that’s giving false readings in HA. I discovered that a couple of zwave in-wall switches aren’t updating their status when manually turned on and off. For example, one model (Nortek Security WS15Z-1) won’t update any manual on-off commands. But through HA or Z2M control panel work fine. Another model I have a few of are Leviton DZ15S-1BZ in wall switches but what’s bizarre about these ones, are they have add-ons for 3 and 4-way configurations. Everything is properly updated if I manually turn on/off the switch from the “main” switch, but any remotes don’t update at all. I have other switches that are working just fine with no issues. I can send OZW logs if needed, but nothing shows up when manually pressing the rocker on those switches. Any thoughts? I tried creating an account with OpenZwave to analyze the logs, but their website is severely broken and the registration process doesn’t work.
